packages/skill-lesson/trpc/routers/pricing.ts

```diff
@@ -43,24 +43,30 @@ const checkForAnyAvailableUpgrades = async ({
     (purchase) => purchase.productId,
   )
 
-  const availableUpgrades = await availableUpgradesForProduct(
+  const potentialUpgrades = await availableUpgradesForProduct(
     validPurchases,
     productId,
   )
 
+  type AvailableUpgrade = Awaited<
+    ReturnType<typeof availableUpgradesForProduct>
+  >[0]
+  // filter out potential upgrades that have already been purchased
+  const availableUpgrades = potentialUpgrades.filter<AvailableUpgrade>(
+    (
+      availableUpgrade: AvailableUpgrade,
+    ): availableUpgrade is AvailableUpgrade => {
+      return !productIdsAlreadyPurchased.includes(
+        availableUpgrade.upgradableTo.id,
+      )
+    },
+  )
```

Comment:
I originally (in yesterday's PR) got the filtering wrong -- we want to filter out an potential upgrade _if_ the `upgradableTo` has already been purchased.